### PR TITLE
resultsdb-updater: add support for redhat-module artifact

### PR DESCRIPTION
**Pull request by thrix**
Also cover with test with a real-life message being sent out.

Signed-off-by: Miroslav Vadkerti <mvadkert@redhat.com>

### CODE REVIEW

**Comment by mikebonnet** [comment]
+1 LGTM!

File: resultsdbupdater/utils.py [hunk 1]
```diff
@@ -330,6 +330,31 @@ def handle_ci_umb(msg):
                 ('category', msg_body.get('category')),
             ) if value is not None
         }
+    elif item_type == 'redhat-module':
+        msg_body_ci = msg_body['ci']
+
+        result_data = {
+            'item': msg_body['artifact']['nsvc'],
```
[the remark this answers is not in the document]
Comment (thrix) [reply resultsdbupdater/utils.py]: @happz @optak we are leaving the delimiter as ```:```

resultdb updater does the transformation

**Comment by thrix** [comment]
@gnaponie should be fixed now

**Comment by gnaponie** [comment]
+1

File: resultsdbupdater/utils.py [hunk 2]
```diff
@@ -330,6 +330,37 @@ def handle_ci_umb(msg):
                 ('category', msg_body.get('category')),
             ) if value is not None
         }
+    elif item_type == 'redhat-module':
+        msg_body_ci = msg_body['ci']
+
+        # The pagure.io/messages spec defines the NSVC delimited with ':', but koji expects:
+        # - NSV delmited with ':'
+        # - C delimited with '.'
+        nsvc = msg_body['artifact']['nsvc'].replace(':', '-', 2).replace(':', '.')
```
Comment (lubomir) [inline resultsdbupdater/utils.py]: The mapping from `N:S:V:C` to `N-S-V.C` is a bit more complicated. Stream name can contain a `-` symbol, and MBS replaces it with `_` when importing to Koji. So `mod:my-stream:20181123:deadcafe` should transform to `mod-my_stream-20181123.deadcafe`.

Comment (thrix) [reply resultsdbupdater/utils.py]: ok, so first replace - with _ and then to the same I did :)

Comment (lubomir) [reply resultsdbupdater/utils.py]: Name can also contain dashes and those should be preserved. You need to split it and `replace('-', '_')` only in the `S` part.

Comment (thrix) [reply resultsdbupdater/utils.py]: ah that would not work, just the stream name ... ok, more code needed then. Thanks @lubomir

**Pull request merged**